app/routes/synonyms/synonyms.server.ts

```diff
@@ -145,7 +151,7 @@ export async function moderatorCreateSynonyms(synonymousEventIds: string[]) {
     RequestItems: {
       [TableName]: synonymousEventIds.map((eventId) => ({
         PutRequest: {
-          Item: { synonymId: uuid, eventId },
+          Item: { synonymId: uuid, eventId, slug: slug(eventId) },
```

Comment:
Hoist `slug(eventId)` to function scope to avoid calling it multiple times.

Reply:
I'm not sure that would work. It occurs once in the moderatorCreateSynonyms() map function. If it was hoisted to the function level, all the synonyms would have one singular slug (based on which?) eventId and that's not consistent with how it works. You would want each individual eventId to be slugged since they are different.

It occurs once in tryInitSynonym(), so no real benefit to hoisting it there. 

it occurs in two map functions in putSynonyms(). Again, each synonym being updated or created would need a unique slug based on its eventId. If hoisted in this function, all updated and created synonyms would have the same slug regardless of different eventIds. 

Can you clarify where and why it should be hoisted to function level? Are you wanting all synonyms in a group to have the same slug regardless of eventId?

Reply:
I see. I simply misread the code. Thanks!